services/export/tests/unit/shared/requestAll.test.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const requestAll = require('../../../shared/requestAll');
+
+describe('Test requestAll method', () => {
+	const func = async (params) => ({
+		data: [params],
+		meta: {
+			total: 1000,
+		},
+	});
+
+	it('should call passed function multiple times when total limit > single response limit', async () => {
+		const singleRequestLimit = 5;
+		const expectedResponse = [
+			{
+				extra_param: 'extra_value',
+				offset: singleRequestLimit * 0,
+				limit: singleRequestLimit,
+			},
+			{
+				extra_param: 'extra_value',
+				offset: singleRequestLimit * 1,
+				limit: singleRequestLimit,
+			},
+			{
+				extra_param: 'extra_value',
+				offset: singleRequestLimit * 2,
+				limit: singleRequestLimit,
+			},
+			{
+				extra_param: 'extra_value',
+				offset: singleRequestLimit * 3,
+				limit: singleRequestLimit,
+			},
+		];
```

Comment:
Could be simplified to and defined as a function to 
```suggestion
		const expectedResponseLength = 4;
		const expectedResponse = getResponseOfLength(expectedResponseLength);
```

where `getResponseOfLength` can be as follows:

```
const getResponseOfLength = (n) => new Array(n).fill().map((e, i) => ({
	extra_param: 'extra_value',
	offset: singleRequestLimit * i,
	limit: singleRequestLimit,
}));
```

Reply:
Applied.